### ISSUE TITLE
Add automated updates of attribute tables

### ISSUE THREAD

**Issue by misaugstad** [ProjectSidewalk/SidewalkWebpage#1267]
In #1266 (resolving issue #1240), clustering of labels into attributes was added. This has to be run manually right now, and recalculates for the entire database, which takes about 15 minutes to run on my laptop, if I remember correctly. This can be automated in two ways

1. There should probably be a nightly, in the same way that we recalculate street priority nightly (PR #1216).
2. We should also be able to real time-ish updates as well. What I would like to do is to update the attribute tables after every audit task that is finished, but this would be too time intensive, especially for users who have placed a lot of labels (since there is a single user clustering step that happens first). We could probably do something like an hourly update, or half-hourly update or something, that
    1. Finds all the users that placed labels in the past hour (or whatever)
    2. Run single-user clustering on just those users (mayyybe we could do it only on the labels from those users that are within the neighborhoods that they audited in in the past hour)
    3. Run multi-user clustering on only the neighborhoods where users have audited in the past hour.

I need to think through the specifics a little more, and we may want to change how the attribute tables are organized a little bit, but this is a sketch of where I'm at on this right now.

**Comment by misaugstad** [comment]
When this happens I think the best way would be to add a new table that holds each `user_id` and whether we consider them a "high quality user". We would then update this table nightly. This table could be used for a lot of things, including improving our real-time updates of street priority.

**Comment by misaugstad** [comment]
I think that table could also have a column that indicates whether that user has been manually marked as a high quality user (which we would likely do when evaluation turker audits).

**Comment by andrew4699** [comment]
The nightly clustering is finished.

@misaugstad What are the criteria for automatically marking high-quality users? And we could just keep that information (along with the "did we mark them manually?") in sidewalk_user.

**Comment by misaugstad** [comment]
We are set up to do something more complicated, but as of right now we are using the following metric: they are a "high quality" user if they have placed at least 3.75 labels per 100 meters they've audited. We perform this calculation when doing the clustering. Starting at this line in `AttributeController.scala`:

`val goodUsers: List[String] = StreetEdgePriorityTable.getIdsOfGoodUsers`

And although we could technically use the `sidewalk_user`, I think I want to use a new table. The `sidewalk_user` table feels like something we should try not to touch. User ID, username, and email address should probably be separate from something like this.

**Comment by andrew4699** [comment]
Is the "high quality user" part of this task separate from the clustering? Or are we using the high quality marking to influence the clustering in some way?

**Comment by misaugstad** [comment]
>Is the "high quality user" part of this task separate from the clustering? Or are we using the high quality marking to influence the clustering in some way?

We use labels from high quality user part in that we only include high quality users in the clustering.

>would we then have 1 row per user in the high quality user table? Or only insert high quality users into that table with a "manual" column?

I think we would have 1 row per user. The table would be called something like `user_quality` or `user_stats`.

>Additionally, how do we handle manually-marked users getting auto-marked later on?

I think that the table would have maybe a nullable column that contains what the users were manually marked as. So `null` means no manual verification, `true` means we manually reviewed and are saying high quality, and `false` means we manually reviewed and said low quality.

The more I think about this, I believe this can be in a separate PR, and you don't necessarily have to spearhead that part. This discussion has been very helpful for clarifying my thoughts on how to implement though.
